horiz_interp/horiz_interp_type.F90

```diff
@@ -181,43 +181,44 @@ subroutine stats ( dat, low, high, avg, miss, missing_value, mask )
  end subroutine stats
 
 !######################################################################################################################
+!> @brief Creates a copy of the horiz_interp_type object
  subroutine horiz_interp_type_eq(horiz_interp_out, horiz_interp_in)
-    type(horiz_interp_type), intent(inout) :: horiz_interp_out
-    type(horiz_interp_type), intent(in)    :: horiz_interp_in
+    type(horiz_interp_type), intent(inout) :: horiz_interp_out !< Output object being set
+    type(horiz_interp_type), intent(in)    :: horiz_interp_in !< Input object being copied
 
     if(.not.horiz_interp_in%I_am_initialized) then
       call mpp_error(FATAL,'horiz_interp_type_eq: horiz_interp_type variable on right hand side is unassigned')
     endif
 
-    horiz_interp_out%faci            => horiz_interp_in%faci
-    horiz_interp_out%facj            => horiz_interp_in%facj
-    horiz_interp_out%ilon            => horiz_interp_in%ilon
-    horiz_interp_out%jlat            => horiz_interp_in%jlat
-    horiz_interp_out%area_src        => horiz_interp_in%area_src
-    horiz_interp_out%area_dst        => horiz_interp_in%area_dst
-    horiz_interp_out%wti             => horiz_interp_in%wti
-    horiz_interp_out%wtj             => horiz_interp_in%wtj
-    horiz_interp_out%i_lon           => horiz_interp_in%i_lon
-    horiz_interp_out%j_lat           => horiz_interp_in%j_lat
-    horiz_interp_out%src_dist        => horiz_interp_in%src_dist
+    horiz_interp_out%faci            =  horiz_interp_in%faci
+    horiz_interp_out%facj            =  horiz_interp_in%facj
+    horiz_interp_out%ilon            =  horiz_interp_in%ilon
+    horiz_interp_out%jlat            =  horiz_interp_in%jlat
+    horiz_interp_out%area_src        =  horiz_interp_in%area_src
+    horiz_interp_out%area_dst        =  horiz_interp_in%area_dst
+    horiz_interp_out%wti             =  horiz_interp_in%wti
+    horiz_interp_out%wtj             =  horiz_interp_in%wtj
+    horiz_interp_out%i_lon           =  horiz_interp_in%i_lon
+    horiz_interp_out%j_lat           =  horiz_interp_in%j_lat
+    horiz_interp_out%src_dist        =  horiz_interp_in%src_dist
```

Comment:
I think you need to check if the variable is allocated before you set it.
```fortran
    if (allocated(horiz_interp_in%wtj)) horiz_interp_out%wtj             =  horiz_interp_in%wtj
```

Reply:
I see what you are saying - do you think it would also be necessary to check if a variable is allocated before allocating?

```
if (allocated(variable)) deallocate(variable)
allocate(variable)
```

Reply:
Tom was right. It will fail if you try to assign an unallocated variable. I tested it in a simple code.

Reply:
@thomas-robinson I wrote a test program to see if we would get an error if `horiz_interp_type_out` is unallocated, and there is no error and the program runs fine.  I found a section of the Fortran 2003 standard that seems to indicate that If a member of horiz_interp_out is unallocated, it will be allocated with the same shape as horiz_interp_in.

You can see the section of the fortran 2003 standard I am referring to [here](https://wg5-fortran.org/N1601-N1650/N1601.pdf#page=155) in Section 7.4.1.3 Interpretation of intrinsic assignments

Reply:
@thomas-robinson Can you take a look at my previous comment on this thread and let me know if you agree with my reasoning?

Reply:
You need to do this for all of the allocatable variables.